theano/sandbox/cuda/blas.py

```diff
@@ -417,13 +417,19 @@ def __hash__(self):
         return hash(type(self)) ^ hash(self.inplace)
 
     def __setstate__(self, dct):
-        inplace = dct.get('inplace', True)
-        if inplace:
+        self.__dict__.update(dct)
+        if self.inplace:
             self.destroy_map = {0: [0]}
-        self.inplace = inplace
+
+        # Correctly reload older pickles where _op_use_c_code was not saved
+        if '_op_use_c_code' not in self.__dict__:
+            self._op_use_c_code = True
 
     def __getstate__(self):
-        return dict(inplace=self.inplace)
+        rval = self.__dict__.copy()
+        # Do not serialize the destroy_map
+        rval.pop('destroy_map', None)
```

Comment:
Why don't we want to serialize destroy_map?  There shouldn't be any problems with pickle.